src/game/Spell.cpp

```diff
@@ -1958,9 +1958,40 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, UnitList&
         }
         case TARGET_SINGLE_FRIEND:
         case TARGET_SINGLE_FRIEND_2:
-            if (m_targets.getUnitTarget())
-                targetUnitMap.push_back(m_targets.getUnitTarget());
+        {
+            if (Unit* unitTarget = m_targets.getUnitTarget())
+            {
+                if (unitTarget->IsFriendlyTo(m_caster))
+                    targetUnitMap.push_back(unitTarget);
+                else
+                {
```

Comment:
That might be true, but recently I got a huge lecture how it prevents bugs due to writing code at 2 AM when you are half sleepy. I am gonna keep it.